web/ui/react-app/src/index.tsx

```diff
@@ -5,15 +5,21 @@ import App from './App';
 import 'bootstrap/dist/css/bootstrap.min.css';
 
 // Declared/defined in public/index.html, value replaced by Prometheus when serving bundle.
-declare const PATH_PREFIX: string;
+declare const GLOBAL_PATH_PREFIX: string;
+declare const GLOBAL_CONSOLES_LINK: string;
 
-let prefix = PATH_PREFIX;
-if (PATH_PREFIX === 'PATH_PREFIX_PLACEHOLDER' || PATH_PREFIX === '/') {
+let prefix = GLOBAL_PATH_PREFIX;
+let consolesLink = GLOBAL_CONSOLES_LINK;
+if (GLOBAL_PATH_PREFIX === 'PATH_PREFIX_PLACEHOLDER' || GLOBAL_PATH_PREFIX === '/') {
   // Either we are running the app outside of Prometheus, so the placeholder value in
   // the index.html didn't get replaced, or we have a '/' prefix, which we also need to
   // normalize to '' to make concatenations work (prefixes like '/foo/bar/' already get
   // their trailing slash stripped by Prometheus).
   prefix = '';
 }
 
-ReactDOM.render(<App pathPrefix={prefix} />, document.getElementById('root'));
+if (GLOBAL_CONSOLES_LINK === 'CONSOLES_LINK_PLACEHOLDER' || GLOBAL_CONSOLES_LINK === '') {
```

Comment:
I think is good to add null/undefined check here as well. Please check https://github.com/prometheus/prometheus/blob/1a1859417696170f2ceaff34222942bd95bbda78/web/ui/react-app/src/utils/index.ts#L14

Reply:
I don't think we need to check null/undefined here, since that would be an error (since it is always set to something in index.html). We don't do it for the path prefix either.

Reply:
If null is coming out from the API then what value will be passed to <App> for consolesLink? I think it's going to be null because line 22 will never be reached and as far I can see the check in the navbar is ```consolesLink !== ''``` which will return true and the navbar item will be rendered incorrectly

Reply:
@boyskila `!isPresent` should be used for both the global value.